lib/filewriter.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * FileWriter -- fast non-streaming file appender for accepting buffered data
+ *
+ * Locks the file when appending to be safe with multiple writers.
+ * Also, it reopens the file periodically, to make it work with logrotate.
+ *
+ * Copyright (C) 2014-2015 Andras Radics
+ * Licensed under the Apache License, Version 2.0
+ */
+
+var fs = require('fs');
+var fse = require('./fs-ext.js');
+
+var flock = fse.flock;
+var flockSync = function( fd, mode ) {
+    try { fse.flockSync(fd, mode); return true }
+    catch (err) { return false }
+}
+
+fse.SEEK_SET = 0;
+fse.SEEK_CUR = 1;
+fse.SEEK_END = 2;
+
+module.exports = FileWriter;
+
+function FileWriter( filename, openmode ) {
+    'use strict';
```

Comment:
yes, I too prefer that.  At the time I wrote this code, the linter I was using (jslint I think)
coplained if the use strict was not function-scope